transport/http/header.go

```diff
@@ -25,44 +25,73 @@ import (
 	"strings"
 
 	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/internal/observability"
+	"go.uber.org/yarpc/yarpcerrors"
 )
 
 // headerConverter converts HTTP headers to and from transport headers.
 type headerMapper struct{ Prefix string }
 
 var (
 	applicationHeaders = headerMapper{ApplicationHeaderPrefix}
+
+	// enforceHeaderRules is a feature flag for a more strict error handling rules.
```

Comment:
can you be more specific here? from the impl, it looks it means we will not attach headers if this value is set to true && isReservedHeaderPrefix returns true? If that is the case, please call it out in the comment

Reply:
Updated this comment